libp2p/6-interfaces.md

```diff
@@ -1,139 +1,48 @@
-│            connection                   │
-└─────────────────────────────────────────┘
-┌───────────────┐┌───────────┐┌───────────┐
-│Transport      ││multistream││ stream    │
-│(TCP, UDP, etc)││           ││ muxer     │
-└───────────────┘└───────────┘│┌ ─ ─ ─ ─ ┐│
-                              │  spdy     │
-                              │└ ─ ─ ─ ─ ┘│
-                              │┌ ─ ─ ─ ─ ┐│
-                              │ multiplex │
-                              │└ ─ ─ ─ ─ ┘│
-                              │┌ ─ ─ ─ ─ ┐│
-                              │ QUIC      │
-                              │└ ─ ─ ─ ─ ┘│
-                              │┌ ─ ─ ─ ─ ┐│
-                              │ others    │
-                              │└ ─ ─ ─ ─ ┘│
-                              └───────────┘
-```
-
-## 6.4 Distributed Record Store
-
-
+`libp2p` is a conglomerate of several protocols working together to offer a common solid interface with talking with any other network addressable process. This is made possible by shimming current exist protocols and implementations through a set of explicit interfaces, from Peer Routing, Discovery, Stream Muxing, Transports, Connections and so on.
 
+## 6.1 libp2p
 
+libp2p, the top module that interfaces all the other modules that make a libp2p instance, must offer an interface for dialing to a peer and plugging in all of the modules (e.g. which transports) we want to support.
 
 
 
+## 6.2 Peer Routing
 
-----------------------------
-OLD 
-The network protocol's interface has two parts:A
-
-1. the _client interface_, for clients (e.g. higher layers of IPFS)
-2. the _service interface_, for remote peers (e.g. other IPFS nodes)
-
-### 4.1 Client Interface
-
-The **Client Interface** is exposed to the higher layers of IPFS. It is the entry point for other parts to open + handle streams.
-
-This type system represents the interface exposed to clients. Actual implementations will likely be more complicated, but they should aim to cover this.
-
-```go
-type PrivateKey interface {
-  PublicKey() PublicKey
+![](https://raw.githubusercontent.com/diasdavid/abstract-peer-routing/master/img/badge.png)
 
-  Sign(data []byte) Signature
-  Decrypt(ciphertext []byte) (plaintext []byte)
-}
+A Peer Routing service offers a way for libp2p Node to find the PeerInfo of another Node, so that it can dial to that node. In it is most pure form, a Peer Routing module should have a interface that given a 'key', a set of PeerInfos are returned.
+See https://github.com/diasdavid/abstract-peer-routing for the interface and tests.
 
-type PublicKey interface {
-  PeerID() PeerID
+## 6.3 Swarm
 
-  Verify(Signature) (ok bool)
-  Encrypt(plaintext []byte) (ciphertext []byte)
-}
+Current interface available and updated at:
 
-// PeerID is a hash of a PublicKey, encoded in multihash
-// It represents the identity of a node.
-type PeerID Multihash
+https://github.com/diasdavid/js-libp2p-swarm#usage
```

Comment:
@jbenet @whyrusleeping would you like to bikeshed a bit these function calls? Same goes for the "abstract-<name>", so that they get written on the spec and making everyone happy